src/action/common/provision_nix.rs

```diff
@@ -8,22 +8,40 @@ use crate::{
     },
     settings::{CommonSettings, SCRATCH_DIR},
 };
+use std::os::unix::fs::MetadataExt as _;
 use std::path::PathBuf;
 
+pub(crate) const NIX_STORE_LOCATION: &str = "/nix/store";
+
 /**
 Place Nix and it's requirements onto the target
  */
 #[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
 #[serde(tag = "action_name", rename = "provision_nix")]
 pub struct ProvisionNix {
-    fetch_nix: StatefulAction<FetchAndUnpackNix>,
-    create_nix_tree: StatefulAction<CreateNixTree>,
-    move_unpacked_nix: StatefulAction<MoveUnpackedNix>,
+    pub(crate) fetch_nix: StatefulAction<FetchAndUnpackNix>,
+    pub(crate) create_nix_tree: StatefulAction<CreateNixTree>,
+    pub(crate) move_unpacked_nix: StatefulAction<MoveUnpackedNix>,
 }
 
 impl ProvisionNix {
     #[tracing::instrument(level = "debug", skip_all)]
     pub async fn plan(settings: &CommonSettings) -> Result<StatefulAction<Self>, ActionError> {
+        if std::path::Path::new(NIX_STORE_LOCATION).exists() {
```

Comment:
Can you make this into its own function with a nice name, so we can be real succinct about understanding what it is doing while reading this code later?